### PR TITLE
Benchmark set elgibility

### PR DESCRIPTION
**Pull request by girazoki**
Add test and benchmark for author-slot-filter

### CODE REVIEW

File: pallets/author-slot-filter/src/weights.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			.saturating_add(T::DbWeight::get().writes(3 as Weight))
+	}
+}
+
+// For backwards compatibility and tests
+impl WeightInfo for () {
+	fn set_eligible() -> Weight {
+		(13_019_000 as Weight)
+			.saturating_add(RocksDbWeight::get().reads(4 as Weight))
+			.saturating_add(RocksDbWeight::get().writes(3 as Weight))
```
Comment (notlesh) [inline pallets/author-slot-filter/src/weights.rs]: Looking at the actual fn, this should really only be doing a single write, am I missing something?

Comment (girazoki) [reply pallets/author-slot-filter/src/weights.rs]: Not sure, but this seems to be the case in any other function where we are only setting a storage value, e.g., https://github.com/PureStake/moonbeam/blob/7e89857cde8b3ed57621bf06186766fbaf80b26b/pallets/parachain-staking/src/lib.rs#L2591.

My theory is that deposit event triggers at least one additional db write

Comment (girazoki) [reply pallets/author-slot-filter/src/weights.rs]: This is what benchmarking outputs:

Raw Storage Info
========
Storage: System Number (r:1 w:0)
Storage: System ExecutionPhase (r:1 w:0)
Storage: System EventCount (r:1 w:1)
Storage: System Events (r:1 w:1)
Storage: AuthorFilter EligibleRatio (r:0 w:1)

Comment (notlesh) [reply pallets/author-slot-filter/src/weights.rs]: I think we need to whitelist some of these. Here's what the node template in substrate currently does:

```rust
            let whitelist: Vec<TrackedStorageKey> = vec![
                // Block Number
                hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef702a5c1b19ab7a04f536c519aca4983ac").to_vec().into(),
                // Total Issuance
                hex_literal::hex!("c2261276cc9d1f8598ea4b6a74b15c2f57c875e4cff74148e4628f264b974c80").to_vec().into(),
                // Execution Phase
                hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef7ff553b5a9862a516939d82b3d3d8661a").to_vec().into(),
                // Event Count
                hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef70a98fdbe9ce6c55837576c60c7af3850").to_vec().into(),
                // System Events
                hex_literal::hex!("26aa394eea5630e07c48ae0c9558cef780d41e5e16056765bc8461851072c9d7").to_vec().into(),
            ];
```

Comment (notlesh) [reply pallets/author-slot-filter/src/weights.rs]: ...hmm, the runtime in this repo (`parachain-template/runtime/src/lib.rs`) already has exactly this. What runtime are you using to benchmark this?

Comment (notlesh) [reply pallets/author-slot-filter/src/weights.rs]: Likely related: https://github.com/PureStake/moonbeam/pull/1339

**Comment by girazoki** [comment]
Updated weights @notlesh with the new whitelisted keys

File: pallets/author-slot-filter/src/weights.rs [hunk 2]
```diff
@@ -58,17 +58,13 @@ pub trait WeightInfo {
...
 	}
 }
 
 // For backwards compatibility and tests
 impl WeightInfo for () {
 	fn set_eligible() -> Weight {
-		(13_019_000 as Weight)
-			.saturating_add(RocksDbWeight::get().reads(4 as Weight))
-			.saturating_add(RocksDbWeight::get().writes(3 as Weight))
+		(13_250_000 as Weight).saturating_add(RocksDbWeight::get().writes(1 as Weight))
```
Comment (notlesh) [inline pallets/author-slot-filter/src/weights.rs:68]: :+1:

**Review by notlesh — APPROVED** [review]
LGTM.

Related: https://github.com/PureStake/moonbeam/pull/1350

I already need to update `nimbus` along with that PR, it would be great if this PR could get merged before then.

**Comment by girazoki** [comment]
I will merge it so that you can take it to your update

**Pull request merged**